checkpoint/checkpoint.go

```diff
@@ -0,0 +1,76 @@
+package checkpoint
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/intel/multus-cni/types"
+)
+
+const (
+	checkPointfile = "/var/lib/kubelet/device-plugins/kubelet_internal_checkpoint"
+)
+
+type PodDevicesEntry struct {
+	PodUID        string
+	ContainerName string
+	ResourceName  string
+	DeviceIDs     []string
+	AllocResp     []byte
+}
+
+type checkpointData struct {
+	PodDeviceEntries  []PodDevicesEntry
+	RegisteredDevices map[string][]string
+}
+
+type Data struct {
+	Data     checkpointData
+	Checksum uint64
+}
+
+// getPodEntries gets all Pod device allocation entries from checkpoint file
+func getPodEntries() ([]PodDevicesEntry, error) {
+
+	podEntries := []PodDevicesEntry{}
+
+	cpd := &Data{}
+	rawBytes, err := ioutil.ReadFile(checkPointfile)
```

Comment:
I haven't seen issue on getting device information from checkpoint. If we start to see that file is not being updated by the time CNI is invoked we definitely can try to re-read the file after some intervals. It is more of deciding how long & how often we should try to re-read the file if we don't see the expected info. Also bear in mind we need to throw in error as soon as we can in case of invalid resourceName given. 
I'd prefer to the tackle this corner case later once we understand the problem well.